File: DSA/Python/BinarySearch/binary_search.py
def search(nums: List[int], target: int) -> int:
    left = 0
    right = len(nums) - 1
    
    while left < right:
        pointer = left + (right-left)//2 #int((right-left)/2)
        
        value = nums[pointer]
        if value == target:
            return pointer
        
        if value < target:
            left = pointer + 1
        else:
            right = pointer - 1
            
    return -1

File: DSA/Dotnet/LeetCode.Net/Problems/BinarySearch/BinarySearch.cs


namespace LeetCode.Problems.BinarySearch;

public class BinarySearch
{
    public int Search(int[] nums, int target) {
        var leftPointer = 0;
        var rightPointer = nums.Length-1;

        while(leftPointer <= rightPointer)
        {
            var pointer = leftPointer + (rightPointer-leftPointer) / 2;

            if(nums[pointer] == target)
            {
                return pointer;
            }

            if(nums[pointer] < target)
            {
                leftPointer = pointer + 1;
                continue;
            } 
            if(nums[pointer] > target)
            {
                rightPointer = pointer - 1;
                continue;
            }
        }

        return -1;
    }
}